### PR TITLE
[WIP] Tape feature parity in templates 

### PR DESCRIPTION
**Pull request by mariaschuld**
Refactors the templates to achieve feature parity when using tape mode.

### CODE REVIEW

File: pennylane/tape/__init__.py [hunk 1]
```diff
@@ -30,6 +30,7 @@
     CVParamShiftTape,
     ReversibleTape,
 )
+from .interfaces import MLFunctionWrapper
```
Comment (mariaschuld) [inline pennylane/tape/__init__.py:33]: I couldn't get any other import method to work!

File: pennylane/tape/interfaces/functions.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+library namespaces/nicknames/interface names to the corresponding wrapper
+class above"""
+
+# Importing this module will result in the dictionary above
+# being imported. E.g.,
+# >>> from pennylane.tape.transforms functions as fn
+# >>> fn["np"]
+# <class NumPyFunctions>
+# >>> fn["np"].abs
+#sys.modules[__name__] = MLFunctionWrapper
```
Comment (mariaschuld) [inline pennylane/tape/interfaces/functions.py:198]: I think this created the circular import trouble.

Comment (josh146) [reply pennylane/tape/interfaces/functions.py:198]: :+1: This was me playing around to see if I could get auto framework detection working 😆 I could never get it to work though, so this is fine to remove!

**Comment by trbromley** [comment]
[ch1036]

**Comment by mariaschuld** [comment]
Opening another PR with a more recent version.

**Pull request closed without merging**